pkg/controllers/streaming/stream_controller.go

```diff
@@ -82,21 +95,244 @@ func (r *StreamReconciler) reconcile(ctx context.Context, logger logr.Logger, st
 	stream.Status.InitializeConditions()
 
 	// delegate to the provider via its REST API
-	logger.Info("calling provisioner for Stream", "provisioner", stream.Spec.Provider)
+	log.Info("calling provisioner for Stream", "provisioner", stream.Spec.Provider)
 	address, err := r.StreamProvisionerClient.ProvisionStream(stream)
 	if err != nil {
 		stream.Status.MarkStreamProvisionFailed(err.Error())
 		return ctrl.Result{Requeue: true}, err
 	}
 	stream.Status.MarkStreamProvisioned()
 	stream.Status.Address = *address
```

Comment:
should we just move the ConfigMap/Secret generation here rather than writing the address into the Status, since its duplicating information in the Secret (especially if we think that information does indeed belong in a Secret)?

Reply:
I think yes eventually, but as an interim migration path, this needs to be kept

Reply:
I'm not worried about the gateway and topic names as sensitive values. Keeping them for backcompt in the moment is useful, we can decide to remove them before shipping.

We should not add actual credentials to the Stream's status, those should only go into the Secret.

Reply:
I still think we should move the ConfigMap/Secret generation rather than having it cascade from the Address. That way it happens where it would if we didn't already have Address (or if we remove it later). I'm still not sold on having the value in both a Secret and directly in the Status. If it's not sensitive, why not add it to the ConfigMap instead of the Secret? Is this a matter of conforming to the CNB spec's distinction between buildtime and runtime as reflected in the `metadata` vs `secret` sub-directories? If so, I think that should be reconsidered.

Reply:
> I still think we should move the ConfigMap/Secret generation rather than having it cascade from the Address.

We can cross that bridge when we get to it. I'd like to build out the capability iteratively.

> If it's not sensitive, why not add it to the ConfigMap instead of the Secret?

Not all values in secrets are sensitive, but all sensitive values should be in secrets.

>  Is this a matter of conforming to the CNB spec's distinction between buildtime and runtime as reflected in the metadata vs secret sub-directories?

The CNB Binding spec doesn't say where an incremental value should go. But there are a couple hints. Here's my current thinking:
1. if the value contains a credential or other sensitive value, it goes under secret
2. if the value is needed at build time it goes under metadata
3. if the value describes what the bound service, it goes under metadata
4. if the value describes how to connect to the bound service, it goes under secret
5. default to the metadata

The current gateway address and topic name fall under item 4, and go into the secret.

Reply:
okay, this sounds fine as long as we have an issue to remove Address from the Status before 0.5